src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -5236,6 +5250,21 @@ private void leaveRing()
         Uninterruptibles.sleepUninterruptibly(delay, MILLISECONDS);
     }
 
+    public boolean isDecommissionFailed()
+    {
+        if (operationMode == Mode.LEAVING && hasDecommissionFailed)
```

Comment:
I think we can change this to `if (hasDecommissionFailed && operationMode == Mode.LEAVING)`
if hasDecommissionFailed is false we will not need to do the comparison with operationMode
Edit :
or why we just return 
hasDecommissionFailed && operationMode == Mode.LEAVING
? and not need for if () else